### ISSUE TITLE
history list not working?

### ISSUE THREAD

**Issue by opto** [opto/nostalgy-xpi#64]
at the moment, the history at my end is just the last list of folders from the previous move.
Maybe I am too tired, but I thought it would show the history of the last n moves.
Anybody else observing this?

**Comment by rogerval** [comment]
Hello from the other side… of the world! :-)

The history seems to be behaving the same way for me, too.

Straight after starting Thunderbird, if I click on a message and type the "save" command I don’t get any history, just a narrow window to type my desired folder.

Next, if I type the first letter of my desired folder, a list of 20 folders pops up (I have it set to remember the last 20 folders), but they don't seem to be the history, but rather 20 folders whose names contain the letter I've typed. (The folder I need isn't necessarily in the list, so I have to type another letter or two before the desired folder appears in the list.)

Next I select the folder required, and save the email in that folder.

Next I click on another email and type the save command and this time I get the last list of folders Nostalgy suggested during the previous save operation. This last list of folders keeps coming back for each save, unless I select a different folder to save to, in which case, _that_ list of folders keeps being presented.

Hope this helps you track down the problem. :-)

(Just for sake of completeness: my environment is: Nostalgy++ 2.2.2, TB 78.5.1 on Windows 10 2004, OS build 19041.630, Lenovo ThinkStation P310)
(My Nostalgy++ settings are: match on folder name only, not full path; use statistical prediction; remember last 20 folders; consider 100 addresses for statistical prediction.)

**Comment by ziomau10** [comment]
Same for me.

**Comment by tonycanike** [comment]
Yes, I see the same issue.  Not getting a history of recently moved-to folders, just the last auto-matched folders.

**Comment by Olifair** [comment]
Finally I also found this as the only issue I have with Nostalgy++ . I have not been able to figure out something systematic,  but very often the results from the last search before will be shown rather than a history of recently used folders. Not a big problem for me, but it would be even more efficient if this would work too. Otherwise I'm really happy with Nostalgy++ . Thanks a lot opto.

**Comment by jaysmithgit** [comment]
Same problem on
Kubuntu 20.04, Thunderbird 78.7.1, Nostalgy++ 2.2.4

However, if I do up or down arrow, I _can_ file into whatever the last-filed-into mailbox was, but only that one. 

It is as if Nostalgy++ is looking at the wrong list, not a filed-history list, but a query-qualifying-history list.  However, nothing (other than the last filed mailbox name) from the list that is displayed is selectable even if one of them were a desired target -- until and unless I start typing all over again to generate a new list.

On old Nostalgy, we made a huge amount of use of this feature.  We file hundreds of emails per day, 70% of them to a core group of approximately 30 mailboxes, thus having access again to the history list would be tremendously helpful.

I hope that you will be able to find the time to make it work.  I will be *very* happy to donate generously -- you can hold me to that -- when it works at least as well as old Nostalgy.

**Comment by rogerval** [comment]
I've seen (in the instructions I think) that if I type the "move" shortcut key, followed by by left arrow then up arrow, I should see the history appear, but that doesn't work for me for some unknown reason.
I noticed yesterday that I can get the folders history to appear by pressing the "move" shortcut key (as installed, "s"), then typing a space, which then causes the archives folder for the first listed mail account to appear in blue after the space character in the folder completion box. Then if I press the backspace key twice, followed by the up arrow, the history list appears above the folder completion box. I can then press the up arrow to highlight the desired folder, and "enter" to cause the message moving to occur.
Hopefully this helps someone (and I hope I haven't just succeded in confusing the issue further :-) )
(Currently using  Nostalgy++ 2.3.9, TB 78.11.0 64 bit on Windows 10 2004, OS build 19041.1052, Lenovo ThinkStation P310)

**Comment by opto** [comment]
In 3.0.6, it shows the last search results (on commands S,M etc.). If a space is entered, it switches to the history of chosen locations.
Just typing on goes into the folder discovery. See https://github.com/opto/nostalgy-xpi/releases/tag/v3.0.6.


This works in 91.0b6 and has not (yet) been tested in 78.

**Comment by rogerval** [comment]
I'm currently running TB 78.12.0 64-bit on Windows 10 (21H1 19043.1110) and I'm pleased to report that the entering of a space character works as expected for me, as does the folder discovery. Thanks for fixing this. :-)
